cpp/src/io/parquet/page_data.cu

```diff
@@ -1666,7 +1674,10 @@ __global__ void __launch_bounds__(block_size) gpuDecodePageData(
         } else if (dtype == INT96) {
           gpuOutputInt96Timestamp(s, val_src_pos, static_cast<int64_t*>(dst));
         } else if (dtype_len == 8) {
-          if (s->ts_scale) {
+          if (s->dtype_len_in == 4) {
+            // Reading INT32 TIME_MILLIS in 64-bit DURATION_MILLISECONDS
```

Comment:
Do we need to link the Parquet specification or something like that? It's not obvious why this choice is made.

Reply:
added link